File: CreApps.StarterKit.Models/Ticket.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CreApps.StarterKit.Models
{
    public class Ticket : EntityBase<int>
    {
        public string Subject { get; set; }
        public int TypeId { get; set; }
        public  TicketType Type { get; set; }
        public int StatusId { get; set; }
        public Status Status { get; set; }
        public int PriorityId { get; set; }
        public Priority Priority { get; set; }
        public string Description { get; set; }
    }
}

File: CreApps.StarterKit.WebSPA/ClientApp/src/Models/Ticket.ts
import { TicketType } from '../Models/TicketType';
import { Status } from '../Models/Status'
import { Priority } from '../Models/Priority'

export class Ticket {
    Subject: string;
    TypeId: number;
    //Type: TicketType;
    StatusId: number;
    //Status: Status;
    PriorityId: number;
    //Priority: Priority;
    Description: string;
}